src/validator.js

```diff
@@ -271,14 +360,22 @@ export default class Validator {
         })
         .then(() => {
           this.xpi = new _Xpi(this.packagePath);
+          return this.detectPackageType();
+        })
+        .then((packageType) => {
+          this.packageType = packageType;
+          if (packageType) {
+            log.debug('Package type detected as %s',
```

Comment:
If there’s an extra benefit to doing the `%s` stuff for logging purposes I’m cool with it. Otherwise just seems easier to be consistent with string usage.